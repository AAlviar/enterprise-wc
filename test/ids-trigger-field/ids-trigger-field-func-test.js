/**
 * @jest-environment jsdom
 */
import IdsTag from '../../src/ids-trigger-field/ids-trigger-field';

describe('IdsTriggerField Component', () => {
  let triggerField;

  beforeEach(async () => {
    const elem = new IdsTriggerField();
    document.body.appendChild(elem);
    triggerField = document.querySelector('.ids-trigger-field');
  });

  afterEach(async () => {
    document.body.innerHTML = '';
  });

  it('renders with no errors', () => {
    const errors = jest.spyOn(global.console, 'error');
    const elem = new IdsTag();
    document.body.appendChild(elem);
    elem.remove();
    expect(document.querySelectorAll('.ids-trigger-field').length).toEqual(1);
    expect(errors).not.toHaveBeenCalled();
  });

});
