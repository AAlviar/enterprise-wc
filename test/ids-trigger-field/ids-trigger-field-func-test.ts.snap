// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdsTriggerField Component renders correctly 1`] = `
"<ids-trigger-field id=\\"trigger-field-1\\" size=\\"sm\\" label=\\"Date Field\\">
      <ids-trigger-button slot=\\"trigger-end\\">
        <ids-text audible=\\"true\\">Date Field trigger</ids-text>
        <ids-icon slot=\\"icon\\" icon=\\"schedule\\"></ids-icon>
      </ids-trigger-button>
    </ids-trigger-field>"
`;

exports[`IdsTriggerField Component renders correctly 2`] = `
"<ids-trigger-field id=\\"trigger-field-1\\" size=\\"sm\\" label=\\"Date Field\\">
      <ids-trigger-button slot=\\"trigger-end\\">
        <ids-text audible=\\"true\\">Date Field trigger</ids-text>
        <ids-icon slot=\\"icon\\" icon=\\"schedule\\"></ids-icon>
      </ids-trigger-button>
    </ids-trigger-field>"
`;
